gordo_components/dataset/base.py

```diff
@@ -111,7 +124,24 @@ def join_timeseries(
             logging.debug(series.head(3))
             logging.debug(series.tail(3))
 
-            resampled = series.resample(resolution, label="left").mean()
+            resampled = series.resample(resolution, label="left").agg(
+                aggregation_methods
+            )
+            # If several aggregation methods are provided, agg returns a dataframe
+            # instead of a series. In this dataframe the column names are the
+            # aggregation methods, like "max" and "mean", so we have to make a
+            # multi-index with the series-name as the top-level and the
+            # aggregation-method as the lower-level index
+            if isinstance(
+                resampled, pd.DataFrame
+            ):  # Several aggregation methods provided
+                resampled.columns = pd.MultiIndex.from_product(
+                    [[series.name], resampled.columns],
+                    names=["tag", "aggregation_method"],
+                )
+            # For backwards-compatibility we *dont* return a multi-level index
+            # when we have a single resampling method.
+
```

Comment:
A little experiment in the terminal seems if there is a multi-index column dataframe, `pd.concat` will, if done with at least one series, will squash mutli-index names into tuples:
```
        (input, col1)  (input, col2)  (input, col3)  (input, col4)  ...  (transformed, col2)  (transformed, col3)  (transformed, col4)  0
0            0.0            1.0            2.0            3.0  ...                  1.5                  3.0                  4.5  0
1            4.0            5.0            6.0            7.0  ...                  7.5                  9.0                 10.5  1
```

I guess this is ok then? I don't actually know, just clarifying.

However, if all are multi-level which I guess which would typically happen, there is no backwards-compatibility because then it does keep the multi-level structure.

Reply:
Hmm. They will all either be single-level or multi-level, since we don't support per-series aggregation method selection, they all either have a single method or all have multiple methods. We could of course could support different aggregation methods, I just did not think about it at all :-p But that can be for the future, just expanding the `aggregation_methods` parameter to also accept a `dict` from tag to resample method(s).

Reply:
I think it's cool as it is now, just adding the check in infrastructure, then it never actually ends up being the multi-level, and we can tackle that later; but I like the idea that this spawned. :+1: